cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -462,6 +467,38 @@ func (csr *ClusterStateRegistry) updateNodeGroupMetrics() {
 	metrics.UpdateNodeGroupsCount(autoscaled, autoprovisioned)
 }
 
+// UpdateSafeScaleUpMetricsForNodeGroup queries the health status and backoff situation of the node group and updates metrics
+func (csr *ClusterStateRegistry) UpdateSafeScaleUpMetricsForNodeGroup(now time.Time) {
```

Comment:
Both the `ForNodeGroup` suffix and the comment above suggest this is about a specific node group, while in fact it iterates over all of them.

Reply:
Modifications done and PR submitted